### PR TITLE
Globally replace "polyline" with "linestring"

### LINKED ISSUE #520: [FEA] `polyline` -> `linestring`?
In general, `polyline` and `linestring` both refers to a data structure that contains multiple points with the line segments that connect the points. `Linestring` is standardized in GIS systems through OGC-SFS and provides a lot more relevant search results in [google search](https://www.google.com/search?q=linestring) as compared to [search with polyline](https://www.google.com/search?q=polyline).

After introduction of #510 there will be two different names of the same concept. I would propose naming `polyline` with `linestring` in the code base to avoid confusion.

### PR DESCRIPTION
**Pull request by harrism**
## Description
Replaces all uses of "polyline" (and abbreviations) globally in cuspatial with "linestring" for consistency.  

Fixes #520

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by isVoid — APPROVED** [review]
Thanks!

**Review by thomcom — CHANGES_REQUESTED** [review]
Looks good, thanks for taking this on. I just noticed one filename that you missed.

File: cpp/src/join/quadtree_poly_filtering.cu [hunk 1]
```diff
@@ -48,7 +48,7 @@ namespace {
 
 template <typename T>
 inline std::unique_ptr<cudf::table> join_quadtree_and_bboxes(cudf::table_view const& quadtree,
```
[the remark this answers is not in the document]
Comment (harrism) [reply cpp/src/join/quadtree_poly_filtering.cu:50]: Good idea.

[the remark this answers is not in the document]
Comment (harrism) [reply cpp/src/join/quadtree_poly_filtering.cu:50]: Done.

**Comment by harrism** [comment]
@gpucibot merge

**Pull request merged**